src/Sarif/Autogenerated/Rule.cs

```diff
@@ -43,7 +43,7 @@ public SarifNodeKind SarifNodeKind
         /// An array of stable, opaque identifiers by which this rule was known in some previous version of the analysis tool.
         /// </summary>
         [DataMember(Name = "deprecatedIds", IsRequired = false, EmitDefaultValue = false)]
-        public IList<string> DeprecatedIds { get; set; }
+        public virtual IList<string> DeprecatedIds { get; set; }
```

Comment:
>IList [](start = 23, length = 5)

Needs to be virtual due to converting skimmer base to derive from Rule (as part of ongoing effort to extirpate IRule)

Reply:
That makes sense. But I'm surprised this the code generator is not marking this property as virtual. Every other property is virtual. What's special about this one? (he asked rhetorically).

---
In reply to: [248136339](https://github.com/Microsoft/sarif-sdk/pull/1215#discussion_r248136339) [](ancestors = 248136339)

Reply:
It could very well be for all I know, Rule.cs is getting picked up from NotYetAutogenerated for some other reason (or unnecessarily). I will take a look and get you a good answer, won't block here.

---
In reply to: [248368566](https://github.com/Microsoft/sarif-sdk/pull/1215#discussion_r248368566) [](ancestors = 248368566,248136339)

Reply:
code gen isn't emitting these as virtual. we may have broken the connection between IRule/Rule that would result in this (as part of our partial extraction process). in which case, ok, expected. or maybe there's a jschema issue. i won't file a bug, because we will remember to go examine and attempt to review the notyetgenerated content as part of fixing other bugs, and we'll fix the blame then where it needs to be.

---
In reply to: [248377693](https://github.com/Microsoft/sarif-sdk/pull/1215#discussion_r248377693) [](ancestors = 248377693,248368566,248136339)